### PR TITLE
[CloudMinds] 10335-Event

### PR DESCRIPTION
**Pull request by Kenny-SZ**
Here lists all items of an event reported by the robot.

### CODE REVIEW

File: 10335.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+<LWM2M xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://openmobilealliance.org/tech/profiles/LWM2M.xsd">
```
Comment (seanmcilroy29) [inline 10335.xml:33]: Invalid XML Schema ref as you've indicated ```<LWM2MVersion>1.1</LWM2MVersion>``` 
Schema should be ```<LWM2M xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.openmobilealliance.org/tech/profiles/LWM2M-v1_1.xsd">```

Comment (jpradocueva) [reply 10335.xml:33]: @Kenny-SZ do you really need Version 1.1? Based on the "Type" used in your object you can use the schema defined for Version 1.0 of the LwM2M protocol.

Comment (Kenny-SZ) [reply 10335.xml:33]: Thanks for your comments, 1.0 is enough.

File: 10335.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    <Resources>
+		<Item ID="1">
+			<Name>Entity</Name>
+			<Operations>R</Operations>
+			<MultipleInstances>Single</MultipleInstances>
+			<Mandatory>Mandatory</Mandatory>
+			<Type>String</Type>
+			<RangeEnumeration>4..63</RangeEnumeration>
+			<Units/>
+			<Description><![CDATA[Contains the object ID and object instance ID/name, for example: &lt;/10320/2&gt;.]]></Description>
```
Comment (akeranen) [inline 10335.xml:52]: This sounds like Object Link type?

File: 10335.xml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		</Item>
+		<Item ID="2">
+			<Name>Event Type</Name>
+			<Operations>R</Operations>
+			<MultipleInstances>Single</MultipleInstances>
+			<Mandatory>Mandatory</Mandatory>
+			<Type>Integer</Type>
+			<RangeEnumeration>0..65535</RangeEnumeration>
+			<Units></Units>
+			<Description><![CDATA[The type of this event, for example: APP restart.]]></Description>
```
Comment (akeranen) [inline 10335.xml:62]: How are the numbers mapped to types?

File: 10335.xml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		</Item>
+		<Item ID="3">
+			<Name>Time</Name>
+			<Operations>R</Operations>
+			<MultipleInstances>Single</MultipleInstances>
+			<Mandatory>Mandatory</Mandatory>
+			<Type>Time</Type>
+			<RangeEnumeration></RangeEnumeration>
+			<Units></Units>
+			<Description><![CDATA[When was this event reported.]]></Description>
```
Comment (akeranen) [inline 10335.xml:72]: There's also re-usable time stamp resource that could be used here

**Review by akeranen — CHANGES_REQUESTED** [review]

File: 10335.xml (hunk 2 above)
Comment (Kenny-SZ) [reply 10335.xml:52]: Yes, if it's object ID + instance ID，then it likes a Objlnk, but when it's Object ID + instance name， then it's not a Objlnk.

For some objects, it has a lot of instances, and the instance ID is a little meaningless, we tends to use instance name to indicate a object instance.
For some objects, we don't define a OMA object for them, we may use  /Computer/Kenny to indicate Kenny's Computer in an event.

File: 10335.xml (hunk 3 above)
Comment (Kenny-SZ) [reply 10335.xml:62]: We have defined dozens of event types, but I didn't list them here.

For example: 3100 is for APP restart, 5000 is for robot restart.

In my opinion, it's not a good way to use one Object ID for each Event or Alarm, we prefer to use only one Object for Event, and use its resource to indicate which Event/Alarm it is.

File: 10335.xml (hunk 4 above)
Comment (Kenny-SZ) [reply 10335.xml:72]: If there is already a reused resource, and the meaning is the same, we can consider to use it, thanks.

**Comment by mkgillmore** [comment]
Group agrees to merge 1/9/2020

**Pull request merged**